Products/PluggableAuthService/plugins/ZODBGroupManager.py

```diff
@@ -307,6 +321,17 @@ def removePrincipalFromGroup( self, principal_id, group_id ):
             self._principal_groups[ principal_id ] = new
             self._invalidatePrincipalCache( principal_id )
 
+            # Get PAS instance to lookup principal object to notify event
+            pas = self._getPAS()
+            # As we don't know if principal is user or group at this point
+            # try to get user object first
+            obj = pas.getUserById(principal_id)
+            # Otherwise get group object
+            if not obj:
+                obj = pas.getGroupById(principal_id)
+            # Notify event
```

Comment:
Factor the common bits out, e.g. into a `_get_principal(principal_id)` method.

Reply:
Good call, I've done that now